examples/firmware/README.md

```diff
@@ -0,0 +1,104 @@
+# TPM Firmware Update Support
+
+Currently wolfTPM supports firmare update capability for the Infineon SLB9672 (SPI) and SLB9673 (I2C) TPM 2.0 modules. Infienon has open sourced their firmware update.
+
+## Infineon Firmware
+
+### Extracting the firmware
+
+Infienon releases firmware as a .bin file (example: TPM20_15.23.17664.0_R1.BIN).
+
+The .bin contains a 16-byte GUID header, at least one manifest based on key group and the firmware. A typical manifest is 3KB and firmware is 920KB.
+
+We have included a host side tool `ifx_fw_extract` for extracting the manifest and firmware data file required for a TPM upgrade.
+
+Example usage:
+
+```sh
+# Build host tool
+make
+
+# Help
+./ifx_fw_extract --help
+Usage:
+  ifx_fw_extract <fw-file>
+  ifx_fw_extract <fw-file> <keygroup_id> <manifest-file> <data-file>
+
+# Find key groups in .bin
+./ifx_fw_extract TPM20_26.13.17770.0_R1.BIN
+Reading TPM20_26.13.17770.0_R1.BIN
+Found group 00000007
+
+# Extract manifest and firmware data files for key group
+./ifx_fw_extract TPM20_26.13.17770.0_R1.BIN 7 TPM20_26.13.17770.0_R1.MANIFEST TPM20_26.13.17770.0_R1.DATA
+Reading TPM20_26.13.17770.0_R1.BIN
+Found group 00000007
+Chosen group found: 00000007
+Manifest size is 3224
+Data size is 934693
+Writing TPM20_26.13.17770.0_R1.MANIFEST
+Writing TPM20_26.13.17770.0_R1.DATA
+```
+
+### Updating the firmware
+
+The `ifx_fw_update` tool uses the manifest (header) and firmware data file.
+
+The TPM has a vendor capability for getting the key group id. This is populated in the `WOLFTPM2_CAPS.keyGroupId` when `wolfTPM2_GetCapabilities` is called. This value should match the firmware extract tool `keygroup_id`.
+
+```sh
+# Help
+./ifx_fw_update --help
+Infineon Firmware Update Usage:
+        ./ifx_fw_update (get info)
+        ./ifx_fw_update --abandon (cancel)
+        ./ifx_fw_update <manifest_file> <firmware_file>
```

Comment:
Usage nitpick: if someone accidentally reverses these 2 args, would it harmlessly error out or do something bad?

Reply:
It will fail the manifest check. It will return a return code 0x8E = `TPM_RC_AUTH_FAIL`